src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,141 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    public static final int DEFAULT_ARRAY_SIZE = 10;
+    private int size;
+
+    private T[] elementData;
+
+    public ArrayList() {
+        this.elementData = (T[]) new Object[DEFAULT_ARRAY_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size >= elementData.length) {
+            elementData = grow();
+        }
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (isEmpty()) {
+            elementData[0] = value;
+            size++;
+            return;
+        }
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        if (checkIndex(index)) {
+            System.arraycopy(elementData, index,
+                    elementData, index + 1,
+                    size - index);
+            elementData[index] = value;
+            size++;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounds");
```

Comment:
Replace all hardcoded messages to constants

Reply:
usually, error messages are not made constants